samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -346,9 +352,10 @@ public void stop() {
         try {
           LOGGER.info("Shutting down the container: {} of stream processor: {}.", container, processorId);
           boolean hasContainerShutdown = stopSamzaContainer();
+          // todo: use interruptSamzaContainer method instead to give one more chance for container to shutdown
```

Comment:
Few reasons
1. I didn't want to change existing behavior for the code path that isn't affected by the bug.
2. Lack of shutting down timely could mean two things, one it didn't react to shutdown signal or it is spending more than `task.shutdown.ms` on the shutdown sequence. Interrupting it would require handling interrupts during shutdown sequence as well and requires more testing. 

let me get rid of the todo for now and create a follow up ticket to evaluate if giving it one more chance has tangible benefits.

Reply:
Sounds good thanks.